packages/harmony/src/components/select/Select.tsx

```diff
@@ -0,0 +1,250 @@
+  options: SelectOption[]
+  optionRefs?: RefObject<HTMLButtonElement[]>
+  onChange: (option: SelectOption) => void
+}
+
+export const SelectOptions = (props: SelectOptionsProps) => {
+  const { activeValue, options, onChange, optionRefs } = props
+  const { color, cornerRadius, spacing, typography } = useTheme()
+
+  // Popup Styles
+  const optionIconCss: CSSObject = {
+    width: spacing.unit4,
+    height: spacing.unit4
+  }
+
+  const activeOptionCss: CSSObject = {
+    transform: 'none',
+    backgroundColor: color.secondary.s300,
+    color: color.special.white
+  }
+
+  const optionCss: CSSObject = {
+    background: 'transparent',
+    border: 'none',
+    color: color.text.default,
+    fontWeight: typography.weight.medium,
+    gap: spacing.s,
+    paddingLeft: spacing.m,
+    paddingRight: spacing.m,
+    paddingTop: spacing.s,
+    paddingBottom: spacing.s,
+    width: '100%',
+    borderRadius: cornerRadius.s,
+    justifyContent: 'flex-start',
+
+    '&:hover': activeOptionCss,
+
+    '&:active': {
+      transform: 'none'
+    }
+  }
+
+  if (!options.length) {
+    return (
+      <Flex justifyContent='center'>
+        <Text variant='body' color='subdued' size='s'>
+          {messages.noMatches}
+        </Text>
+      </Flex>
+    )
+  }
+
+  return (
+    <Flex direction='column'>
+      {options.map((option, index) => (
+        <BaseButton
+          key={option.value}
+          iconLeft={option.icon}
+          styles={{
+            button: {
+              ...optionCss,
+              ...(option.value === activeValue ? activeOptionCss : {})
+            },
+            icon: optionIconCss
+          }}
+          onClick={() => onChange(option)}
+          aria-label={option.label ?? option.value}
+          role='option'
+          ref={(el) => {
+            if (optionRefs && optionRefs.current && el) {
+              optionRefs.current[index] = el
+            }
+          }}
+        >
+          {option.leadingElement ?? null}
+          <Text variant='body' strength='strong'>
+            {option.label ?? option.value}
+          </Text>
+          {option.helperText ? (
+            <Text variant='body' strength='strong' color='subdued'>
+              {option.helperText}
+            </Text>
+          ) : null}
+        </BaseButton>
+      ))}
+    </Flex>
+  )
+}
+
+export const Select = forwardRef<HTMLInputElement, SelectProps>(function Select(
+  props,
+  ref
+) {
+  const {
+    selection: selectionProp,
+    options,
+    optionsLabel,
+    popupAnchorOrigin = { horizontal: 'left', vertical: 'bottom' },
+    popupMaxHeight,
+    popupTransformOrigin = { horizontal: 'left', vertical: 'top' },
+    popupPortalLocation,
+    popupZIndex,
+    ...selectInputProps
+  } = props
+
+  const [selection, setSelection] = useControlled({
+    controlledProp: selectionProp,
+    defaultValue: null,
+    stateName: 'selection',
+    componentName: 'FilterButton'
+  })
+
+  // TODO: implement filtering
```

Comment:
Looks like this is done?

Reply:
No not entirely, let me update the comment